### PR TITLE
HLRC: Add ML get overall buckets API

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
Relates #29827

### CODE REVIEW

**Comment by dimitris-athanasiou** [comment]
retest this please

File: client/rest-high-level/src/test/java/org/elasticsearch/client/ml/GetBucketsRequestTests.java [hunk 1]
```diff
@@ -18,47 +18,36 @@
...
 
-import org.elasticsearch.client.ml.job.util.PageParams;
+import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.test.AbstractXContentTestCase;
 
 import java.io.IOException;
 
-public class GetBucketsRequestTests extends AbstractXContentTestCase<GetBucketsRequest> {
+public class GetBucketsRequestTests extends AbstractXContentTestCase<GetOverallBucketsRequest> {
```
Comment (droberts195) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/ml/GetBucketsRequestTests.java]: It looks like you duplicated this file then edited the original rather than the duplicate.  The newly added `GetOverallBucketsRequestTests.java` contains the test for `GetBucketsRequest` and the pre-existing `GetBucketsRequestTests.java` contains the test for `GetOverallBucketsRequest`.

Comment (dimitris-athanasiou) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/ml/GetBucketsRequestTests.java]: Good catch!

File: docs/java-rest/high-level/ml/get-buckets.asciidoc [hunk 2]
```diff
@@ -70,7 +70,7 @@ include-tagged::{doc-tests}/MlClientDocumentationIT.java[x-pack-ml-get-buckets-s
 
 ["source","java",subs="attributes,callouts,macros"]
 --------------------------------------------------
-include-tagged::{doc-tests}/MlClientDocumentationIT.java[x-pack-ml-get-buckets-end]
+include-tagged::{doc-tests}/MlClientDocumentationIT.java[x-pack-ml-get-buckets-start]
```
Comment (droberts195) [inline docs/java-rest/high-level/ml/get-buckets.asciidoc]: I guess it's not really part of this PR, and it's correct that the comment below applies to start rather than end, but there is no reference in the docs to the block tagged `x-pack-ml-get-buckets-end` in the test file.

**Review by droberts195 — APPROVED** [review]
LGTM

I noticed one possible omission from the existing get_buckets docs based on a correction you made, but it's not really part of this PR so feel free to leave any fix for another day.

File: docs/java-rest/high-level/ml/get-buckets.asciidoc (hunk 2 above)
Comment (dimitris-athanasiou) [reply docs/java-rest/high-level/ml/get-buckets.asciidoc]: I'd rather fix it in this one. I'll add a commit.

Comment (dimitris-athanasiou) [reply docs/java-rest/high-level/ml/get-buckets.asciidoc]: Actually, just noticed it is referred. You might have missed it as they are listed alphabetically.
